Fix #12

joncloud opened the pull request:
Added a public Random property on Scientist, which is initialized to a better implementation and thread-safe calculation.

Haacked commented:
Nice! Let's go even simpler. Let's not make it a property of Scientist and just use this one private implementation.

joncloud commented:
Sounds good.  Should be good to go once appveyor picks it up.

joncloud commented:
Actually I think I misinterpreted you.  I'll need to move that to the `ExperimentInstance` class from `Scientist`.

Haacked commented:
Yeah, I think it should be a static private member of `ExperimentInstance`.

Haacked commented:
Have you been following the discussion in #12?

joncloud commented:
Yeah I should be able to address that no problem.

joncloud commented:
I reverted back to `Random`, and added a locking object to ensure thread-safety.  Also I integrated the .net core changes from master to ensure build compatibility.

Haacked commented on src/Scientist/Internals/ExperimentInstance.cs:
```diff
@@ -60,8 +61,14 @@ internal ExperimentInstance(string name, NamedBehavior control, NamedBehavior ca
 
             // Randomize ordering...
+            NamedBehavior[] orderedBehaviors;
+            lock (_randomLock)
```
Not a big deal, but since `_random` is a private instance and not shared anywhere, we could have just locked on `_random` itself.

Haacked commented:
[![selfie-3](http://i.imgur.com/PNbWHnL.gif)](https://github.com/thieman/github-selfies/)
